src/font.c

```diff
@@ -50,6 +50,36 @@ void font_create_ctfont(struct font* font) {
   CTFontDescriptorRef descriptor = CTFontDescriptorCreateWithAttributes(attr);
 
   if (font->ct_font) CFRelease(font->ct_font);
+
+
+  if (font->features) {
+    CFAllocatorRef default_allocator = CFAllocatorGetDefault();
+    
+    char* features_copy = string_copy(font->features);
+    char* feature = strtok(features_copy, ",");
+
+    while (feature) {
+      int feat_name = 0;
+      int feat_value = 0;
+
+      if (sscanf(feature, "%d:%d", &feat_name, &feat_value) == 2) {
+        CFNumberRef name = CFNumberCreate(default_allocator, kCFNumberIntType, &feat_name);
+        CFNumberRef value = CFNumberCreate(default_allocator, kCFNumberIntType, &feat_value);
+
+        descriptor = CTFontDescriptorCreateCopyWithFeature(descriptor, name, value);
```

Comment:
I believe this causes a memory leak, because it copies the existing descriptor and then overwrites the pointer.